server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -582,6 +585,19 @@ export class SummaryWriter implements ISummaryWriter {
         pending: ISequencedOperationMessage[]): Promise<ISequencedDocumentMessage[]> {
         if (lt - gt <= 1) {
             return [];
+        } else if (pending.length > 0) {
+            // Check for logtail ops in the unprocessed ops currently present in memory
+            const pendingOps = pending.filter((op) => op.operation.sequenceNumber > gt && op.operation.sequenceNumber < lt).map((op) => op.operation);
+            if (pendingOps.length > 0 && pendingOps[0].sequenceNumber === gt + 1 && pendingOps[pendingOps.length - 1].sequenceNumber === lt - 1) {
+                Lumberjack.info(`LogTail of length ${pendingOps.length} between ${gt+1} and ${lt-1} fetched from pending ops`, this.lumberProperties);
+                return pendingOps;
+            }
+        }
+        
+        let logTail: ISequencedDocumentMessage[] = [];
+
+        if (this.getDeltasViaAlfred) {
+            logTail = await this.deltaService.getDeltas("", this.tenantId, this.documentId, gt, lt);
```

Comment:
>getDeltas [](http://example.com/codeflow?start=46&length=9)

Does getDeltas() fall back to storage? does getDeltas() return false if it can only get part of required deltas?

Reply:
Correct. This gets the deltas using the alfred api.